### PR TITLE
Add RBAC support

### PR DESCRIPTION
**Pull request by viniciusccarvalho**
Fixes spring-cloud/spring-cloud-deployer-kubernetes#162
Fixes spring-cloud/spring-cloud-dataflow-server-kubernetes#243

### CODE REVIEW

**Comment by sabbyanandan** [comment]
Hi, @trisberg: When you get a chance, it'd be great if you can weigh-in on the proposed changes.

**Comment by trisberg** [comment]
I'm getting a merge conflict on the docs that I'm not entirely sure how to resolve, could you rebase the PR?

**Comment by viniciusccarvalho** [comment]
@trisberg just fixed it

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 1]
```diff
@@ -138,10 +138,25 @@ The ConfigMap settings are specified in the `src/kubernetes/server/server-config
 NOTE: We are now configuring the Data Flow server with file based security and the default user is 'user' with a password of 'password'. Feel free to change this in the `src/kubernetes/server/server-config-rabbit.yaml` file.
 +
 NOTE: The default memory for the pods is set to 1024Mi.  Update the value in the `src/kubernetes/server/server-deployment.yaml` file if you expect most of your apps to require more memory.
+=======
```
Comment (trisberg) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: Formatting is off starting here

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 2]
```diff
@@ -138,10 +138,25 @@ The ConfigMap settings are specified in the `src/kubernetes/server/server-config
...
+We haven't tuned the memory use of the OOTB apps yet, so to be on the safe side we are increasing the memory for the pods by providing the following environment variable in the `src/kubernetes/server/server-deployment.yaml` file:
++
+```
+        - name: SPRING_CLOUD_DEPLOYER_KUBERNETES_MEMORY
+          value: 640Mi
+```
+IMPORTANT: The latest releases of kubernetes have enabled https://kubernetes.io/docs/admin/authorization/rbac/[RBAC] on the api-server. If your target platform has RBAC enabled you must ask a `cluster-admin` to create the `roles` and `role-bindings` for you before deploying the dataflow server. They associate the dataflow service account with the roles it needs to be run with.
+```
+$ kubectl create -f src/kubernetes/server/server-roles.yaml
+$ kubectl create -f src/kubernetes/server/server-rolebindings.yaml
```
Comment (trisberg) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: file name is `server-rolebinding.yaml`

File: src/kubernetes/server/server-roles.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: scdf-role
+  namespace: default
+rules:
+  - apiGroups: [""]
+    resources: ["services", "pods","replicationcontrollers"]
```
Comment (trisberg) [inline src/kubernetes/server/server-roles.yaml]: nit-pick: space after comma

File: src/kubernetes/server/server-roles.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    verbs: ["get", "list", "watch", "create", "delete", "update"]
+  - apiGroups: [""]
+    resources: ["configmaps", "secrets"]
+    verbs: ["get", "list", "watch"]        
+  - apiGroups: ["extensions"]
+    resources: ["deployments", "replicasets","jobs"]
```
Comment (trisberg) [inline src/kubernetes/server/server-roles.yaml]: another space after comma

**Review by trisberg — CHANGES_REQUESTED** [review]
Other than the doc changes and minor formatting issues LGTM, works, tried simple stream and task

**Comment by trisberg** [comment]
Updated to use BUILD-SNAPSHOT versions and getting a failure when deploying ticktock stream

```
2017-12-20 19:30:39.965 ERROR 1 --- [p-nio-80-exec-6] o.s.c.d.s.s.AppDeployerStreamDeployer    : Exception when deploying the app [stream name = ticktock, application name = log, application properties = {spring.metrics.export.triggers.application.includes=integration**, spring.cloud.dataflow.stream.app.label=log, spring.cloud.stream.metrics.key=ticktock.log.${spring.cloud.application.guid}, spring.cloud.stream.bindings.input.group=ticktock, spring.cloud.stream.metrics.properties=spring.application.name,spring.application.index,spring.cloud.application.*,spring.cloud.dataflow.*, spring.cloud.stream.bindings.applicationMetrics.destination=metrics, spring.cloud.dataflow.stream.name=ticktock, spring.cloud.dataflow.stream.app.type=sink, spring.cloud.stream.bindings.input.destination=ticktock.time}: Create StatefulSet failed. response code 403, message Forbidden
```

**Comment by trisberg** [comment]
This configuration seems to work with StatefulSets:

```
kind: Role
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: scdf-role
  namespace: default
rules:
  - apiGroups: [""]
    resources: ["services", "pods", "replicationcontrollers", "persistentvolumeclaims"]
    verbs: ["get", "list", "watch", "create", "delete", "update"]
  - apiGroups: [""]
    resources: ["configmaps", "secrets"]
    verbs: ["get", "list", "watch"]
  - apiGroups: ["apps"]
    resources: ["statefulsets"]
    verbs: ["get", "list", "watch", "create", "delete", "update"]
  - apiGroups: ["extensions"]
    resources: ["deployments", "replicasets", "jobs"]
    verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
```

**Comment by viniciusccarvalho** [comment]
Just squashed all the commits @trisberg let me know if it addresses all issues

**Comment by trisberg** [comment]
LGTM, merging ...

**Comment by trisberg** [comment]
Merged as commit 9a568b153c49e65cb2e04c09a03ca7c3011524bf

**Pull request closed without merging**